arborist/client.go

```diff
@@ -0,0 +1,181 @@
+package arborist
+
+import (
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq"
+)
+
+type Client struct {
+	Name     string   `json:"name"`
+	Policies []string `json:"policies"`
+}
+
+type ClientFromQuery struct {
+	Name     string         `db:"name"`
+	Policies pq.StringArray `db:"policies"`
+}
+
+func (clientFromQuery *ClientFromQuery) standardize() Client {
+	client := Client{
+		Name:     clientFromQuery.Name,
+		Policies: clientFromQuery.Policies,
+	}
+	return client
+}
+
+func clientWithName(db *sqlx.DB, name string) (*ClientFromQuery, error) {
+	stmt := `
+		SELECT
+			client.name,
+			array_remove(array_agg(policy.name), NULL) AS policies
+		FROM client
+		LEFT JOIN client_policy ON client.id = client_policy.client_id
+		LEFT JOIN policy ON policy.id = client_policy.policy_id
+		WHERE client.name = $1
+		GROUP BY client.id
+		LIMIT 1
+	`
+	clients := []ClientFromQuery{}
+	err := db.Select(&clients, stmt, name)
+	if len(clients) == 0 {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	client := clients[0]
+	return &client, nil
+}
+
+func listClientsFromDb(db *sqlx.DB) ([]ClientFromQuery, error) {
+	stmt := `
+		SELECT
+			client.name,
+			array_remove(array_agg(policy.name), NULL) AS policies
+		FROM client
+		LEFT JOIN client_policy ON client.id = client_policy.client_id
+		LEFT JOIN policy ON policy.id = client_policy.policy_id
+		GROUP BY client.id
+	`
+	clients := []ClientFromQuery{}
+	err := db.Select(&clients, stmt)
+	if err != nil {
+		return nil, err
+	}
+	return clients, nil
+}
+
+func (client *Client) createInDb(db *sqlx.DB) *ErrorResponse {
```

Comment:
Still missing granting `client.Policies` to the newly-created client. As this was copied from `user.go`, I guess the same applies for `user.Groups` and `user.Policies`.